tests/cupy_tests/core_tests/test_ndarray.py

```diff
@@ -181,3 +204,107 @@ def test_output_type_mismatch(self, xp):
         i = testing.shaped_arange(self.indices, xp, numpy.int32) % 3
         o = testing.shaped_arange(self.out_shape, xp, numpy.float32)
         wrap_take(a, i, out=o)
+
+
+@testing.parameterize(
+    *testing.product({
+        'indices_shape': [(2,), (2, 2)],
+        'axis': [None, 0, 1, 2, -1, -2],
+    })
+)
+@testing.gpu
+class TestNdarrayPut(unittest.TestCase):
+
+    shape = (4, 4, 5)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_put(self, xp, dtype):
+        a = xp.zeros(self.shape, dtype=dtype)
+        if self.axis is None:
+            m = a.size
+        else:
+            m = a.shape[self.axis]
+        i = testing.shaped_arange(self.indices_shape, xp, numpy.int32) % m
+        v_shape = compute_v_shape(self.shape, self.indices_shape, self.axis)
+        v = testing.shaped_arange(v_shape, xp, dtype)
+        wrap_put(a, i, v, self.axis)
+        return a
+
+
+@testing.parameterize(
+    *testing.product({
+        'indices': [2, [0, 1], -1, [-1, -2]],
+        'axis': [None, 0, 1, -1, -2],
+    })
+)
+@testing.gpu
+class TestNdarrayPutWithInt(unittest.TestCase):
+
+    shape = (3, 4, 5)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_put(self, xp, dtype):
+        a = xp.zeros(self.shape, dtype=dtype)
+        v_shape = compute_v_shape(
+            self.shape, numpy.array(self.indices).shape, self.axis)
+        v = testing.shaped_arange(v_shape, xp, dtype)
+        wrap_put(a, self.indices, v, self.axis)
+        return a
+
+
+@testing.parameterize(
+    *testing.product({
+        'indices': [0, -1, [0]],
+        'axis': [None, 0, -1],
+    })
+)
+@testing.gpu
+class TestScalaNdarrayPutWithInt(unittest.TestCase):
+
+    shape = ()
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_put(self, xp, dtype):
+        a = xp.zeros(self.shape, dtype=dtype)
+        v_shape = compute_v_shape(
+            self.shape, numpy.array(self.indices).shape, self.axis)
+        v = testing.shaped_arange(v_shape, xp, dtype)
+        wrap_put(a, self.indices, v, self.axis)
+        return a
+
+
+@testing.parameterize(
+    {'shape': (3, 4, 5), 'indices': (2,), 'axis': 0, 'v_shape': (5,)},
+)
```

Comment:
We need not to use parameterized test if there is one combination of parameters.